File: java/src/main/java/mx/iteso/stores/MilkIceCream.java
package mx.iteso.stores;

import mx.iteso.IceCreamStore;
import mx.iteso.icecreams.IceCream;
import mx.iteso.icecreams.milk.StrawberryMilkIceCream;
import mx.iteso.icecreams.milk.GuavaMilkIceCream;
import mx.iteso.icecreams.milk.MangoMilkIceCream;
import mx.iteso.icecreams.milk.PineappleMilkIceCream;
import mx.iteso.icecreams.milk.NoFlavorMilkIceCream;

/**
 * Para hacer llamada del helado de leche.
 */
public class MilkIceCream extends IceCreamStore {
    /**
     * Create ice cream.
     * @param selectedFlavor the desired flavor.
     * @return IceCream.
     */
    @Override
    protected IceCream iceCreamFlavor(final String selectedFlavor) {
        if (selectedFlavor.equals("Guava")) {
            return new GuavaMilkIceCream();
        } else if (selectedFlavor.equals("Mango")) {
            return new MangoMilkIceCream();
        } else if (selectedFlavor.equals("Pineapple")) {
            return new PineappleMilkIceCream();
        } else if (selectedFlavor.equals("Strawberry")) {
            return new StrawberryMilkIceCream();
        }
        return new NoFlavorMilkIceCream();
    }
}

File: csharp/factory-method/MilkIceCream.cs
﻿using factorymethod.main.icecreams;
using factorymethod.main.icecreams.milk;

namespace factorymethod.main.stores
{
    internal class MilkIceCreamStore : IceCreamStore
    {
        override
            protected IceCream createIceCream(string flavor)
        {
            if (flavor.Equals("Strawberry"))
            {
                return new StrawberryMilkIceCream();
            }

            if (flavor.Equals("Pineapple"))
            {
                return new PineappleMilkIceCream();
            }
            if (flavor.Equals("Mango"))
            {
                return new MangoMilkIceCream();
            }
            return flavor.Equals("Guava") ? new GuavaMilkIceCream() : null;
        }

    }
}
}
